### PR TITLE
Jcf Set / it-academy.by / JC1-110-22 / Alena Trafimovich / Done tasks 1-3

### PR DESCRIPTION
**Pull request by AlenaTrafimovich**
## Pull request state

Actual status for PR:
- [x] implemented
- [x] reviewed
- [x] fixed
- [ ] approved

### CODE REVIEW

File: src/main/java/com/rakovets/course/java/core/practice/jcf_set/TaxService.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
+package com.rakovets.course.java.core.practice.jcf_set;
+
+import java.util.*;
+
+public class TaxService {
+    private final Map<Integer, Set<Penalty>> taxPayers = new HashMap<>();
+
+    public Set<Penalty> addOrGetPersonInfo(int personalCode) {
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/jcf_set/TaxService.java:8]: Bad design... 2 responsibility:

* add
* get

File: src/main/java/com/rakovets/course/java/core/practice/jcf_set/TaxService.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        if (!taxPayers.containsKey(personalCode)) {
+            taxPayers.put(personalCode, new HashSet<>());
+        }
+        return getTaxPayersByPersonalCode(personalCode);
+    }
+
+    public Penalty addPenalty(int personalCode, Penalty penalty) {
+        Set<Penalty> penalties = addOrGetPersonInfo(personalCode);
+        return penalties.add(penalty) ? penalty : null;
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/jcf_set/TaxService.java:17]: Don't return `null`

File: src/test/java/com/rakovets/course/java/core/practice/jcf_set/TaxServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.*;
+
+public class TaxServiceTest {
+    private final static Penalty SPEED_GDANSK = new Penalty("Speed", "Gdansk");
+    private final static Penalty SPEED_GDYNIA = new Penalty("Speed", "Gdynia");
+    private final static Penalty TECH_INSPECTION_GDYNIA = new Penalty("Technical Inspection", "Gdynia");
+    private final static Penalty TECH_INSPECTION_GDANSK = new Penalty("Technical Inspection", "Gdansk");
```
Comment (rakovets) [inline src/test/java/com/rakovets/course/java/core/practice/jcf_set/TaxServiceTest.java:13]: Best way: duplicate those values in each tests

**Review by rakovets — CHANGES_REQUESTED** [review]
Need to fix

File: src/main/java/com/rakovets/course/java/core/practice/jcf_set/TaxService.java [hunk 4]
```diff
@@ -0,0 +1,87 @@
+package com.rakovets.course.java.core.practice.jcf_set;
+
+import java.util.*;
```
[the remark this answers is not in the document]
Comment (AlenaTrafimovich) [reply src/main/java/com/rakovets/course/java/core/practice/jcf_set/TaxService.java]: fixed

File: src/main/java/com/rakovets/course/java/core/practice/jcf_set/TaxService.java (hunk 2 above)
Comment (AlenaTrafimovich) [reply src/main/java/com/rakovets/course/java/core/practice/jcf_set/TaxService.java]: fixed

File: src/test/java/com/rakovets/course/java/core/practice/jcf_set/TaxServiceTest.java (hunk 3 above)
Comment (AlenaTrafimovich) [reply src/test/java/com/rakovets/course/java/core/practice/jcf_set/TaxServiceTest.java]: fixed

**Comment by AlenaTrafimovich** [comment]
fixed

**Pull request closed without merging**